lib/oxide-vpc/src/engine/gateway/mod.rs

```diff
@@ -275,3 +282,87 @@ impl Display for VpcMeta {
         write!(f, "vpc-meta")
     }
 }
+
+fn make_holepunch_rules(
```

Comment:
n00b Q probably: would it make sense to implement this set of functions for `IpCidr` instead of as standalone functions? Also, maybe cidr.rs should be it's own module inside `gateway` (for clarity).

Reply:
Ordinarily, yes. There is a bit of a distinction here between `opte(-api)` and `oxide-vpc` though, in that the former is meant to be a platform for different VPC designs (see `NetworkImpl`). The previous code-owner (rpz) wrote a [really nice overview of this split](https://github.com/oxidecomputer/opte/blob/master/lib/opte/README.adoc).

The hole-punching behaviour is specific to our VPC design (and to work with how our gateway layer is structured), which is why I've placed it in the `oxide-vpc` crate -- sadly, it's freestanding because we can't impl on a foreign type. However, moving these to a submodule is very much doable! 😄

Reply:
Reorganised into `gateway/transit.rs` in [2680fda](https://github.com/oxidecomputer/opte/pull/490/commits/2680fda4acefbce29009f2518c7f7a0ea8374764).